tests/test_components.py

```diff
@@ -481,37 +541,80 @@ def test_parse(self):
-            },
-            workflows={"wf4": workflow, "wf5": workflow},
-        )
-        component = Collections(bootstrap)
-        component.raw_data = {"enable": True}
+    def setUp(self):
+        self.user = UserFactory(username="user1")
+        self.wf1 = Workflow.objects.create(name="Workflow 1")
+        self.version = PageVersionFactory()
+        self.page1 = self.version.content.page
+        self.bootstrap = Mock(users={"user1": self.user}, workflows={"wf1": self.wf1}, pages={"page1": self.page1})
+        self.component = Collections(self.bootstrap)
+
+    def test_parse(self):
+        """
+        Test that parsing is able to call the Collection.get_or_create method
+        """
+        component = self.component
+        component.raw_data = {'collection1': {'pages': ['page1'], 'name': 'Collection 1', 'user': 'user1', 'workflow': 'wf1'}}            
+
+        with patch.object(
+            component, 'get_version', return_value=self.version
+        ) as get_version, patch.object(
+            component, 'get_or_create'
+        ) as get_or_create, patch.object(
+            component, 'add_version'
+        ) as add_version:
+            component.parse()
+            get_or_create.assert_called_once_with({'workflow': self.wf1, 'name': 'Collection 1', 'author': self.user})
+            
+    def test_get_version(self):
+        version = self.component.get_version(self.page1)
+        self.assertEqual(self.version, version)
+
+    def test_get_or_create(self):
+        # test add
+        collection_data = {
+            "author": self.user,
+            "workflow": self.wf1,
+            "name": 'Collection 1',
+        }
+        collection = self.component.get_or_create(collection_data)
+        self.assertEqual(collection.id, 1)
+        collection_data2 = {
+            "author": self.user,
+            "workflow": self.wf1,
+            "name": 'Collection 2',
+        }
+        collection2 = self.component.get_or_create(collection_data2)
+        self.assertEqual(collection2.id, 2)
+
+        # test get
+        collection3 = self.component.get_or_create(collection_data)
+        self.assertEqual(collection.id, 1)
+
+    def test_add_version(self):
+        """
+        Test that adding a version to a collection creates 
+        a moderation request which references that version
+        """
+        collection_data = {
+            "author": self.user,
+            "workflow": self.wf1,
+            "name": 'Collection 1',
+        }
+        collection = self.component.get_or_create(collection_data)
+        version = self.component.add_version(collection, self.version)
+        moderation_request = collection.moderation_requests.first()
+        stored_version = moderation_request.version
+        # check the stored version
+        self.assertEqual(self.version, stored_version) 
+        # check the stored content
+        self.assertEqual(moderation_request.version.content, self.version.content)
+        
```

Comment:
The three tests above would not be necessary at all if those methods weren't so strangely isolated out in the code or even if `parse` was well tested. I know I usually ask for more tests, but honestly IMO each test is code that has to be maintained and has to be run (therefore slowing down the test suite), so each test has to give you something concrete. If it doesn't, it shouldn't be there.